Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -14,9 +14,51 @@
 
 #include "SantaDecisionManager.h"
 
+// This is a made-up KAUTH_FILEOP constant which represents a
+// KAUTH_VNODE_WRITE_DATA event that gets passed to SantaDecisionManager's
+// FileOpCallback method.  The KAUTH_FILEOP_* constants are defined in
+// sys/kauth.h and run from 1--7.  KAUTH_VNODE_WRITE_DATA is already defined as
+// 4 so it overlaps with the other KAUTH_FILEOP_* constants and can't be used.
+// We define KAUTH_FILEOP_WRITE as something much greater than 7.
+#define KAUTH_FILEOP_WRITE 100
+
 #define super OSObject
 OSDefineMetaClassAndStructors(SantaDecisionManager, OSObject);
 
+# pragma mark Monitoring PIDs
+
+// This keeps track of all pids associated with compiler processes.  It is defined as a global
+// variable so that the pid monitor threads can access it without needing to reference
+// our instance of SantaDecisionManager.
+static SantaCache<bool> *compiler_pid_set_ = new SantaCache<bool>(500, 5);
+
+// Function to monitor for process termination and then remove the process pid
+// from cache of compiler pids.
+static void pid_monitor(void *param, __unused wait_result_t wait_result) {
+  pid_t pid = (pid_t)(uintptr_t)param;
+  struct timespec ts = { .tv_sec = 1, .tv_nsec = 0 }; // wait 1 sec between each poll
+
+  while (true) {
```

Comment:
If `compiler_pid_set_` goes away we should break the look and kill the thread.

Reply:
Done.